sentry_sdk/client.py

```diff
@@ -391,6 +391,9 @@ def capture_event(
             envelope = Envelope(headers=headers)
 
             if is_transaction:
+                if "profile" in event_opt:
+                    envelope.add_profile(event_opt["profile"])
```

Comment:
```suggestion
                    envelope.add_profile(event_opt.pop("profile"))
```
and remove the `del`